classes/clean_moodle_url.php

```diff
@@ -357,6 +370,13 @@ public static function unclean($clean) {
                 self::log("Rewritten to: $path");
             }
 
+        } else if (preg_match("/^\/user\/(\w+)\/(\w+)$/", $path, $matches)) {
```

Comment:
This is too loose, the last part needs to match /discussions exactly, as it is will match a whole bunch of other unrelated urls